### PR TITLE
CHG: [aml] get player clock to codec via interface

### PR DESCRIPTION
**Pull request by koying**
@FernetMenta cf https://github.com/FernetMenta/xbmc/pull/310
Hope it's more-or-less what you had in mind regarding an interface

/cc @codesnake @stefansaraev

### CODE REVIEW

**Comment by stefansaraev** [comment]
so far so good. now pvr.stalker works great on both wetek play and wetek core. pvr.vdr.vnsi still fails on wetek play - it's like one frame per 5 seconds: http://sprunge.us/ecLU but I am not sure if this is AML specific

still crashing on pvr channel switch while playing in full screen: http://sprunge.us/KYjW

thanks much for taking care.

**Comment by koying** [comment]
Yep. Codec dies before renderer :(
That is now a problem (was not when using static callbacks)

**Comment by MrMC** [comment]
yep, you have to account for that behavior :)

File: xbmc/cores/VideoPlayer/Interfaces/IVPClockCallback.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+class CDVDClock;
+
+class IVPClockCallback
+{
+public:
+  virtual CDVDClock* GetClock() = 0;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Interfaces/IVPClockCallback.h]: It is more robust if we don't pass the entire clock object. This interface should implement methods like GetClock. This way I did it for AE: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/AudioEngine/Interfaces/AEStream.h#L34

File: xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp [hunk 2]
```diff
@@ -90,13 +91,13 @@ class CDVDMsgAudioCodecChange : public CDVDMsg
 };
 
 
-CVideoPlayerAudio::CVideoPlayerAudio(CDVDClock* pClock, CDVDMessageQueue& parent)
+CVideoPlayerAudio::CVideoPlayerAudio(IVPClockCallback* pIVPClock, CDVDMessageQueue& parent)
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp]: we need to keep this if the interface only implements the functions need by the codec.

**Comment by stefansaraev** [comment]
what's the plan here? this, at least brings aml/linux back to life..

**Comment by FernetMenta** [comment]
see my comments

**Comment by koying** [comment]
Cool... 
It's only been 3 days and we are talking pre-alpha, here. 
As apparently I'm on my own, it'll'come'when I've time, ie probably not in the middle of my last working week of the year...

**Comment by koying** [comment]
Updated. @stefansaraev Mind retesting, please?

**Comment by stefansaraev** [comment]
yep. not crashing anymore.

**Comment by FernetMenta** [comment]
thanks, looks good.

**Comment by koying** [comment]
@stefansaraev Crashing in the middle of the stream? Or when stopping it?

**Comment by stefansaraev** [comment]
in tme middle of screen. I start live tv channel. go fullscreen and it crashes after 1 to 3 minutes.

**Comment by koying** [comment]
jenkins build and merge

**Comment by koying** [comment]
Unrelated win32 addons failure

**Pull request merged**